### PR TITLE
Dont show "view child data to level" column if not advanced mode or FF is off

### PR DESCRIPTION
**Pull request by AddisonDunn**
## Product Description

JIRA: [USH-4554](https://dimagi.atlassian.net/browse/USH-4554?atlOrigin=eyJpIjoiYTM3Njg0OGE3MGFiNGUzMDk4MzM3MjdiZDZmZWMzZjciLCJwIjoiaiJ9)

Only displays the "View child data to level" column on the "organization structure" page if the FF is on and the "advanced mode" checkbox is checked.

This is the column in question:

Screenshot 2024-06-04 at 2 04 59 PM

## Technical Summary

Pretty straightforward change I think.

Original PR, for context: #34427

## Feature Flag

USH_RESTORE_FILE_LOCATION_CASE_SYNC_RESTRICTION

### Safety story

- Straightforward change
- Just UI/HTML
- Tested out different scenarios locally, checking/unchecking advanced mode, the FF on/off, seeing that the dropdown still disables properly when "view child data" is turned off

### QA Plan

Don't think this change warrants QA

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

[USH-4554]: https://dimagi.atlassian.net/browse/USH-4554?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: corehq/apps/locations/templates/locations/location_types.html [hunk 1]
```diff
@@ -157,15 +158,17 @@ <h2>{% trans "Organization Levels" %}</h2>
...
-              <th>
-                {% trans "View Child Data to Level" %}
-                <span class="hq-help-template"
-                      data-title="{% trans "View Child Data to Level" %}"
-                      data-content="{% blocktrans %}
-                          <p>Only include cases that show up in the user's restore file down to this level (inclusive).</p>
-                      {% endblocktrans %}">
-                </span>
-              </th>
+              {% if location_case_sync_restriction_enabled %}
```
Comment (esoergel) [inline corehq/apps/locations/templates/locations/location_types.html:161]: You can actually do this in the template without needing to pass anything into context
```suggestion
              {% if request|toggle_enabled:"USH_RESTORE_FILE_LOCATION_CASE_SYNC_RESTRICTION" %}
```

Comment (esoergel) [reply corehq/apps/locations/templates/locations/location_types.html]: Yeah, that's my rationale - whichever requires less code.  Certainly not wrong either way though.

**Review by esoergel — APPROVED** [review]

File: corehq/apps/locations/templates/locations/location_types.html [hunk 2]
```diff
@@ -157,15 +157,17 @@ <h2>{% trans "Organization Levels" %}</h2>
...
-                {% trans "View Child Data to Level" %}
-                <span class="hq-help-template"
-                      data-title="{% trans "View Child Data to Level" %}"
-                      data-content="{% blocktrans %}
-                          <p>Only include cases that show up in the user's restore file down to this level (inclusive).</p>
-                      {% endblocktrans %}">
-                </span>
-              </th>
+              {% if request|toggle_enabled:"USH_RESTORE_FILE_LOCATION_CASE_SYNC_RESTRICTION" %}
+                <th data-bind="visible: advanced_mode">
```
Comment (minhaminha) [inline corehq/apps/locations/templates/locations/location_types.html:161]: Why do you need both the feature flag and another boolean to display this section? Wouldn't one or the other be enough? (Same question applies for below)

Comment (esoergel) [reply corehq/apps/locations/templates/locations/location_types.html:161]: The `advanced_mode` flag is a checkbox that users enable to show more columns:
image

Comment (minhaminha) [reply corehq/apps/locations/templates/locations/location_types.html:161]: gotcha, thanks for explaining yall!

**Review by minhaminha — APPROVED** [review]

**Pull request merged**